### PR TITLE
Add Model.find and Model.remove

### PR DESCRIPTION
**Pull request by mccraveiro**

### CODE REVIEW

**Comment by mccraveiro** [comment]
@brenolf review

File: src/Model.js [hunk 1]
```diff
@@ -51,83 +47,68 @@ export class Model {
...
-    // TODO
-  }
-
-  static limit() {
-    // TODO
+/*eslint-disable no-loop-func*/
+for (let method of Query.methods) {
+  Model.prototype.constructor[method] = function (...args) {
+    let query = new Query(this)
+    return query[method].apply(query, args)
```
Comment (brenolf) [inline src/Model.js:111]: Why not using `arguments` instead?

File: src/Query.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  }
+
+  limit(limit) {
+    this.options.limit = limit
+    return this
+  }
+
+  async count(query = {}) {
+    Object.assign(this.query, query)
+
```
Comment (brenolf) [inline src/Query.js:25]: :scissors:

File: src/Query.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    return await this.Model.collection.count(this.query)
```
Comment (brenolf) [inline src/Query.js:26]: :heart:  :+1:

File: src/Query.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  }
+
+  async find(query = {}, options = {}) {
+    Object.assign(this.query, query)
+    Object.assign(this.options, options)
+
+    let documents = await this.Model.collection.find(this.query, this.options)
+    for (let i in documents) { documents[i] = new this.Model(documents[i]) }
```
Comment (brenolf) [inline src/Query.js:34]:
``` js
documents = documents.map(d => new this.Model(d))
```

File: test/Factory.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    let model = new Models[modelName]()
+    model = Object.assign(model, properties)
+    return await model.save()
+  }
+
+  static async createList(modelName, size, properties) {
+    let list = Array(size);
+    for (let i = 0; i < size; i++) {
+      list[i] = this.create(modelName, properties)
+    }
```
Comment (brenolf) [inline test/Factory.js:17]:
``` js
let list = [ for (i of Array(size)) this.create(modelName, properties) ]
```

File: src/Query.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    return documents
+  }
+
+  async findOne(query = {}, options = {}) {
+    Object.assign(this.query, query)
+    Object.assign(this.options, options)
+
+    let document = await this.Model.collection.findOne(this.query, this.options)
+    if (document) { return new this.Model(document) }
+    else { return null }
```
Comment (brenolf) [inline src/Query.js:44]: Inline conditions? :disappointed:

**Pull request closed without merging**